### PR TITLE
Conversation folders list milestone 1

### PR DESCRIPTION
**Pull request by johnxnguyen**
## What's new in this PR?

This PR allows the user to navigate to the folders conversation list and see all conversations organized into the two default folders, "groups" and "one to ones". Incoming connection requests are displayed at the top of the list as per usual. 

I've also cleaned up the adapters by moving common functionality to the base `ConversationListAdapter` class, so each concrete subclass needs only to define how it accepts and organizes its data.

This PR unfortunately **does not contain functionality to expand/collapse folders** (and neither this state preservation), however there is some preliminary work for supporting this. Namely, I've redirected all click events on the rows through methods on the `ConversationListAdapter`. The base implementation will publish these events to the streams as usual, but now subclasses have the opportunity to respond to the events themselves.

## Notes

There is likely some opportunities to optimize the list loading. The previous implementation would call `notifiyDataSetChanged()` whenever any of the data changed, and this is still the case. However I think we could perhaps be a little smarter about how we update the list. I'll address this in a subsequent PR when I complete the expand/collapse functionality.

#### APK
[Download build #139](http://10.10.124.11:8080/job/Pull%20Request%20Builder/139/artifact/build/artifact/wire-dev-PR2340-139.apk)
[Download build #140](http://10.10.124.11:8080/job/Pull%20Request%20Builder/140/artifact/build/artifact/wire-dev-PR2340-140.apk)
[Download build #143](http://10.10.124.11:8080/job/Pull%20Request%20Builder/143/artifact/build/artifact/wire-dev-PR2340-143.apk)

### CODE REVIEW

**Comment by johnxnguyen** [comment]
build again

File: app/src/main/scala/com/waz/zclient/conversationlist/adapters/ConversationFolderListAdapter.scala [hunk 1]
```diff
@@ -17,61 +17,40 @@
...
 class ConversationFolderListAdapter extends ConversationListAdapter with DerivedLogTag {
 
-  // TODO: pull data from service when ready.
-  private val folders = Seq.empty[Folder]
-  private var items = Seq.empty[Item]
+  def setData(incoming: Seq[ConvId], groups: Seq[ConversationData], oneToOnes: Seq[ConversationData]): Unit = {
+    val folders = Seq(
+      Folder("Groups", groups.map(data => Item.Conversation(data)).toList),
+      Folder("One to One", oneToOnes.map(data => Item.Conversation(data)).toList)
+    )
```
Comment (gizemb) [inline app/src/main/scala/com/waz/zclient/conversationlist/adapters/ConversationFolderListAdapter.scala]: we should get folder names from string resources.

File: app/src/main/scala/com/waz/zclient/conversationlist/adapters/ConversationListAdapter.scala [hunk 2]
```diff
@@ -49,23 +101,50 @@ object ConversationListAdapter {
...
-      view.conversationId = convsAndUsers._1.headOption.map(_.id)
-      view.setIncomingUsers(convsAndUsers._2)
-    }
+  abstract class ConversationRowViewHolder(row: ConversationListRow, listener: RowClickListener)
+    extends RecyclerView.ViewHolder(row)
+      with View.OnClickListener
+      with View.OnLongClickListener {
+
+    row.setOnClickListener(this)
+    row.setOnLongClickListener(this)
```
Comment (gizemb) [inline app/src/main/scala/com/waz/zclient/conversationlist/adapters/ConversationListAdapter.scala]: you can use itemView field of the RecyclerView.ViewHolder here, it is public & final. Then you won't need to pass this ConversationListRow

File: app/src/main/scala/com/waz/zclient/conversationlist/adapters/NormalConversationListAdapter.scala [hunk 3]
```diff
@@ -17,62 +17,17 @@
...
-  // `FIXME: This data could be simplified. We only care about the number of user ids, and
-  // we will only user the first conversation data object in order to present the ConnectRequestFragment
-  // when the incoming row is tapped. I think it would be better to just pass an int here, and handle
-  // the tap differently to make the ConnectRequestFragment appear.
-  private var incomingRequests = (Seq.empty[ConversationData], Seq.empty[UserId])
-
-  def setData(convs: Seq[ConversationData], incoming: (Seq[ConversationData], Seq[UserId])): Unit = {
-    conversations = convs
-    incomingRequests = incoming
+    items ++= convs.map { data => Item.Conversation(data) }
```
Comment (gizemb) [inline app/src/main/scala/com/waz/zclient/conversationlist/adapters/NormalConversationListAdapter.scala]: don't you need to clear this "items" first? It seems like it keeps adding to existing list.

**Comment by johnxnguyen** [comment]
@gizemb I've addressed your requests.

**Pull request merged**